jd4/case.py

```diff
@@ -256,20 +257,58 @@ def read_legacy_cases(config, open):
                           memory_bytes,
                           int(score_str))
 
-def read_yaml_cases(config, open):
-    for case in yaml.safe_load(config)['cases']:
-        if 'judge' not in case:
-            yield DefaultCase(partial(open, case['input']),
-                              partial(open, case['output']),
-                              parse_time_ns(case['time']),
-                              parse_memory_bytes(case['memory']),
-                              int(case['score']))
+def read_auto_cases(open, zip_file, time_limit='1s', memory_limit='128m', judge=''):
+    prefix = ''
+    cases = []
+    for filename in zip_file.namelist():
+        match = RE_INPUT_FILENAME.match(filename)
+        if match:
+            prefix = match.group(1)
+            cases.append(int(match.group(2)))
+    if not cases:
+        raise FormatError('No test case found.')
+    cases.sort()
+    score = 100 // len(cases)                # The basic score of each case
+    divider = len(cases) - 100 % len(cases)  # Decides after which case there's an extra score
```

Comment:
We can have a better name for this variable.

Reply:
I don't know which is better.
Use a,b instead? (fog